src/text-buffer.coffee

```diff
@@ -1199,19 +1200,33 @@ class TextBuffer
   ###
 
   # Public: Save the buffer.
-  save: ->
-    @saveAs(@getPath())
+  save: (options) ->
+    @saveAs(@getPath(), options)
 
   # Public: Save the buffer at a specific path.
   #
   # * `filePath` The path to save at.
-  saveAs: (filePath) ->
+  saveAs: (filePath, options) ->
     unless filePath then throw new Error("Can't save buffer with no file path")
 
     @emitter.emit 'will-save', {path: filePath}
     @emit 'will-be-saved', this if Grim.includeDeprecatedAPIs
     @setPath(filePath)
-    @file.writeSync(@getText())
+
+    if options?.backup and @file.existsSync()
+      backupFilePath = filePath + '~'
+      backupFilePath += '~' while fs.existsSync(backupFilePath)
+      fs.writeFileSync(backupFilePath, @file.readSync())
+
+    try
+      @file.writeSync(@getText())
+      fs.removeSync(backupFilePath) if backupFilePath?
+    catch error
+      if backupFilePath?
+        fs.writeFileSync(filePath, fs.readFileSync(backupFilePath))
```

Comment:
And, of course, before removing the backup file, you need to ensure the main file is fsync'd to disk.  Same technique as above: fsync the data, and fsync the directory if the file was just created.